.github/workflows/ci.yml

```diff
@@ -19,23 +19,124 @@ concurrency:
   cancel-in-progress: ${{ github.event_name != 'workflow_call' }}
 
 jobs:
+  planner:
+    runs-on: ubuntu-latest
+    outputs:
+      jobs_to_run: ${{ steps.plan.outputs.jobs_to_run }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Plan jobs to run
+        id: plan
+        run: |
+          # Fetch base ref for PRs and merge_group events
+          if [ "${{ github.event_name }}" = "pull_request" ] || [ "${{ github.event_name }}" = "merge_group" ]; then
+            git fetch origin ${{ github.base_ref }} --depth=1
+            git diff --name-only origin/${{ github.base_ref }} ${{ github.sha }} > changed_files.txt
+          else
+            # For workflow_dispatch or other events, run all jobs
+
+            jobs=static-analysis,rust,kotlin,swift,website,codeql,build-artifacts,build-perf-artifacts
+            echo "jobs_to_run=$jobs" >> $GITHUB_OUTPUT
+
+            exit 0;
+          fi
+
+          echo "Changed files:"
+          cat changed_files.txt
```

Comment:
```suggestion
          cat changed_files.txt
          
          if grep -q '^\.github/' changed_files.txt; then
            echo "jobs_to_run=$jobs"
            exit 0;
          fi
          
          if grep -q '^\.tool-versions' changed_files.txt; then
            echo "jobs_to_run=$jobs"
            exit 0;
          fi
          
          if [[ "${{ github.ref_name }}" =~ ^bust ]]; then
            echo "jobs_to_run=$jobs"
            exit 0;
          fi
```

Reply:
Feel free to just push a commit!

Reply:
Might be good to add a naming convention so that if we start our branches with `bust` or something, it will trigger all jobs.

Reply:
:thumbsup

Reply:
Hah, looks like you can't delete comments that start with `:`. Oh GitHub... what have you become.